packages/zoe/src/contracts/vpool-xyk-amm/doublePool.js

```diff
@@ -0,0 +1,197 @@
+// @ts-check
+
+import { Far } from '@agoric/marshal';
+import { AmountMath } from '@agoric/ertp';
+import { assert, details as X } from '@agoric/assert';
+import { makeFeeRatio } from '../constantProduct/calcFees';
+import {
+  pricesForStatedInput,
+  pricesForStatedOutput,
+} from '../constantProduct/calcSwapPrices.js';
+
+// Price calculations and swap using a pair of pools. Both pools map between RUN
+// and some collateral. We arrange the trades so collateralInPool will have
+// collateral added and collateralOutPool subtracted. When traders specify an
+// input price, that brand will be the inPool; when they specify the output
+// price that brand is the outPool.
+
+const publicPrices = prices => {
+  return { amountIn: prices.swapperGives, amountOut: prices.swapperGets };
+};
+
+/**
+ * @param {ContractFacet} zcf
+ * @param {XYKPool} collateralInPool
+ * @param {XYKPool} collateralOutPool
+ * @param {bigint} protocolFee
+ * @param {bigint} poolFee
+ * @param {ZCFSeat} feeSeat
+ * @returns {VPool}
+ */
+export const makeDoublePool = (
+  zcf,
+  collateralInPool,
+  collateralOutPool,
+  protocolFee,
+  poolFee,
+  feeSeat,
+) => {
+  const inCentral = collateralInPool.getCentralAmount();
+  const inSecondary = collateralInPool.getSecondaryAmount();
+
+  const outCentral = collateralOutPool.getCentralAmount();
+  const outSecondary = collateralOutPool.getSecondaryAmount();
+
+  const inAllocation = { Central: inCentral, Secondary: inSecondary };
+  const outAllocation = { Central: outCentral, Secondary: outSecondary };
+
+  const centralBrand = inCentral.brand;
+  const centralFeeRatio = makeFeeRatio(poolFee, centralBrand);
```

Comment:
In the doc comment for `makeFeeRatio`, how about a gloss for "basis points" as "hundredths of a percent"? I had to look it up :)

What prompted me to wonder is a mis-match between `bigint` and `Ratio` in a declaration for `provideVPool`.

Reply:
Or even a type synonym `BasisPoint` for `bigint` with the gloss there.